session/session.go

```diff
@@ -98,8 +98,10 @@ func newZKSession(servers string, recvTimeout time.Duration, logger stdLogger, c
 	var event zookeeper.Event
 	select {
 	case event = <-events:
-		if event.State != zookeeper.STATE_CONNECTED {
+		if event.State == zookeeper.STATE_AUTH_FAILED || event.State == zookeeper.STATE_EXPIRED_SESSION {
 			return nil, ErrZKSessionNotConnected
+		} else {
+			break
```

Comment:
No, it's not connected yet. It may be in the process of connecting. It could take a second between the `connecting` and `connected` events, and you don't want the `manage` goroutine to start until you're absolutely certain that you're connected.